pkg/driver/controller.go

```diff
@@ -602,6 +602,7 @@ func (cs *controller) CreateSnapshot(
 		// the capacity of the snapshot will be set as same as the capacity of the
 		// origin volume so that overflow does not occur and snapshot is not dropped
 		WithCapacity(vol.Spec.Capacity).
+		WithThinProvision(vol.Spec.ThinProvision).
```

Comment:
@prateekpandey14 , we are copying the volume spec into the snapshot spec [here](https://github.com/openebs/lvm-localpv/pull/102/files#diff-4e4f8ad2f057d794bd5842ed50e358451e6e1b3b20e5f2836289665bdfbf5625R617). So  I think this is not needed.